### PR TITLE
Modification on printing buttons in views

### PR DESCRIPTION
**Pull request by TimLai125**

### CODE REVIEW

File: report_account_nrq/models/__init__.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Rooms For (Hong Kong) Limited T/A OSCG
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
```
Comment (yostashiro) [inline report_account_nrq/models/__init__.py]: We can actually omit these two lines in __init__.py file (which has been my recent practice following how it's done in OCA).

File: report_account_nrq/models/invoice_print.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Rooms For (Hong Kong) Limited T/A OSCG
```
Comment (yostashiro) [inline report_account_nrq/models/invoice_print.py]: 2016-2017 Quartile Limited

File: report_account_nrq/models/invoice_print.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
```
Comment (yostashiro) [inline report_account_nrq/models/invoice_print.py]: Let's switch the license to LGPL.

File: report_account_nrq/models/invoice_print.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+from odoo import api, fields, models
+
+class InvoicePrint(models.Model):
+    _inherit = 'account.invoice'
+
+    @api.multi
+    def invoice_print(self):
+        self.ensure_one()
+        self.sent = True
+        return self.env['report'].get_action(self, 'report_account_nrq.report_invoice_acceptance')
```
Comment (yostashiro) [inline report_account_nrq/models/invoice_print.py]: This line is too long.  Follow PEP8 styling of 79-char maximum length.

File: report_purchase_nrq/__init__.py [hunk 5]
```diff
@@ -1 +1,5 @@
 # -*- coding: utf-8 -*-
+# Copyright 2017 Quartile Limited
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from . import models
```
Comment (yostashiro) [inline report_purchase_nrq/__init__.py]: Finish with blank line.

File: report_purchase_nrq/models/__init__.py [hunk 6]
```diff
@@ -0,0 +1,5 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Quartile Limited
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from . import purchase_print
```
Comment (yostashiro) [inline report_purchase_nrq/models/__init__.py]: Finish with blank line.

File: report_purchase_nrq/models/purchase_print.py [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+
+from odoo import api, fields, models
+
+class PurchasePrint(models.Model):
+    _inherit = 'purchase.order'
+
+    @api.multi
+    def print_quotation(self):
+        self.write({'state': "sent"})
+        return self.env['report'].get_action(self, 'report_purchase_nrq.report_purchaseorder')
```
Comment (yostashiro) [inline report_purchase_nrq/models/purchase_print.py]: Line too long.

File: report_purchase_nrq/models/purchase_print.py [hunk 8]
```diff
@@ -0,0 +1,13 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Rooms For (Hong Kong) Limited T/A OSCG
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
```
Comment (yostashiro) [inline report_purchase_nrq/models/purchase_print.py]: LGPL

File: report_purchase_nrq/models/purchase_print.py [hunk 9]
```diff
@@ -0,0 +1,13 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Rooms For (Hong Kong) Limited T/A OSCG
```
Comment (yostashiro) [inline report_purchase_nrq/models/purchase_print.py]: 2016-2017 Quartile

File: report_sale_nrq/__init__.py [hunk 10]
```diff
@@ -1,3 +1,5 @@
 # -*- coding: utf-8 -*-
 # Copyright 2016 Rooms For (Hong Kong) Limited T/A OSCG
```
Comment (yostashiro) [inline report_sale_nrq/__init__.py]: 2017-2017 Quartile Limited

File: report_sale_nrq/__init__.py [hunk 11]
```diff
@@ -1,3 +1,5 @@
...
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from . import models
```
Comment (yostashiro) [inline report_sale_nrq/__init__.py]: Finish with a blank line.

File: report_sale_nrq/__init__.py [hunk 12]
```diff
@@ -1,3 +1,5 @@
 # -*- coding: utf-8 -*-
 # Copyright 2016 Rooms For (Hong Kong) Limited T/A OSCG
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
```
Comment (yostashiro) [inline report_sale_nrq/__init__.py]: LGPL

File: report_sale_nrq/__manifest__.py [hunk 13]
```diff
@@ -4,7 +4,7 @@
 {
     "name": "Sales Document Print",
     "summary": "",
-    "version": "10.0.1.0.0",
+    "version": "10.0.1.1.0",
     "category": "Sales",
     "website": "https://www.odoo-asia.com/",
     "author": "Rooms For (Hong Kong) Limited T/A OSCG",
```
Comment (yostashiro) [inline report_sale_nrq/__manifest__.py]: Quartile Limited

File: report_sale_nrq/report/sale_report_quotation.xml [hunk 14]
```diff
@@ -145,4 +145,13 @@
...
         report_type="qweb-pdf"/>
 
+    <report
+        id="sale.report_sale_order"
+        string="Quotation / Order"
+        model="sale.order"
+        report_type="qweb-pdf"
+        menu="False"
+        file="sale.report_saleorder"
+        name="sale.report_saleorder"/>
```
Comment (yostashiro) [inline report_sale_nrq/report/sale_report_quotation.xml:155]: Could this just be like below I wonder?

```
    <report
        id="sale.report_sale_order"
        menu="False"/>
```

**Review by yostashiro — CHANGES_REQUESTED** [review]

File: report_account_nrq/models/__init__.py (hunk 1 above)
Comment (TimLai125) [reply report_account_nrq/models/__init__.py]: How about line 1? Should we keep it?

Comment (yostashiro) [reply report_account_nrq/models/__init__.py]: yes, we should keep it.

File: report_sale_nrq/report/sale_report_quotation.xml (hunk 14 above)
Comment (TimLai125) [reply report_sale_nrq/report/sale_report_quotation.xml:155]: I will test it later

File: report_sale_nrq/models/sale_print.py [hunk 15]
```diff
@@ -0,0 +1,15 @@
...
+
+from odoo import api, fields, models
+
+
+class SalePrint(models.Model):
+    _inherit = 'sale.order'
+
+    @api.multi
+    def print_quotation(self):
+        self.filtered(lambda s: s.state == 'draft').write({'state': 'sent'})
```
Comment (yostashiro) [inline report_sale_nrq/models/sale_print.py]: Line too long.

**Review by yostashiro — APPROVED** [review]

**Pull request merged**